File: PHP/Sayuran.php
<?php
require_once __DIR__ . "/Tanaman.php";

class Sayuran extends Tanaman {
    private string $kategori;
    private int $masaPanen;
    private string $kandunganGizi;

    public function __construct(string $id = "", string $nama = "", string $asal = "", string $kat = "", int $panen = 0, string $gizi = "") {
        parent::__construct($id, $nama, $asal);
        $this->kategori = $kat;
        $this->masaPanen = $panen;
        $this->kandunganGizi = $gizi;
    }

    public function getKategori(): string { return $this->kategori; }
    public function setKategori(string $kat): void { $this->kategori = $kat; }

    public function getMasaPanen(): int { return $this->masaPanen; }
    public function setMasaPanen(int $panen): void { $this->masaPanen = $panen; }

    public function getKandunganGizi(): string { return $this->kandunganGizi; }
    public function setKandunganGizi(string $gizi): void { $this->kandunganGizi = $gizi; }
}
?>

File: CPP/Sayuran.cpp
#include <iostream>
#include <string>
#include "Tanaman.cpp" // Include file .cpp sesuai modul

using namespace std;

class Sayuran : public Tanaman {
private:
    string kategori;
    int masa_panen;
    string kandungan_gizi;

public:
    Sayuran(string id_t = "", string nama_t = "", string asal = "", 
            string kat = "", int panen = 0, string gizi = "")
        : Tanaman(id_t, nama_t, asal) {
        this->kategori = kat;
        this->masa_panen = panen;
        this->kandungan_gizi = gizi;
    }

    string get_kategori() { return kategori; }
    void set_kategori(string kat) { this->kategori = kat; }

    int get_masa_panen() { return masa_panen; }
    void set_masa_panen(int panen) { this->masa_panen = panen; }

    string get_kandungan_gizi() { return kandungan_gizi; }
    void set_kandungan_gizi(string gizi) { this->kandungan_gizi = gizi; }
};